access-front-door/README.md

```diff
@@ -1,3 +1,70 @@
 # Access Front Door
 
-This is the code required for opening the front door latch.
\ No newline at end of file
+This is the code required for opening the front door latch.
+
+
+## Setting up environment variables
+
+1. Copy or rename `access-front-door/src/env.example.py` to `access-front-door/src/env.py`
+2. Set `WIFI_SSID` and `WIFI_PASSWORD` to your wifi ssid and password
+3. Update any other variables
+
+
+## Micropython type hints in VSCode
+
+### Install these extensions:
+* https://marketplace.visualstudio.com/items?itemName=ms-python.python
+* https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance
+
+
+### Setup virtualenv and install stubs
+```bash
+# setup virtualenv
+python3 -m venv .venv
+
+# activate virtualenv
+source ./.venv/bin/activate
+
+# install micropython stubs
+pip3 install -U micropython-esp32-stubs
+
+# install mpremote (optional, lets you connect via command line)
+pip3 install -U mpremote
```

Comment:
The guide I followed had me install `ampy` (https://pypi.org/project/adafruit-ampy/) -- is tgere a difference between it and mpremote? (I've no strong feelings for either, just curious!)

Reply:
I have never heard of `ampy`. `mpremote` is made by micropython

Reply:
I've used `mpremote` before on a mac but it was with a pico, not a esp32